ldms/src/ldmsd/ldmsd_request.c

```diff
@@ -6742,3 +6763,80 @@ static int set_default_authz_handler(ldmsd_req_ctxt_t reqc)
 
 	return 0;
 }
+
+extern struct option long_opts[];
+extern char *short_opts;
+extern int ldmsd_process_cmd_line_arg(char opt, char *value);
+static int __cmdline_options(ldmsd_req_ctxt_t reqc, int argc, char *argv[])
+{
+	int opt, opt_idx;
+	reqc->errcode = 0;
+	opterr = 0;
+	optind = 0;
+	while (0 < (opt = getopt_long(argc, argv,
+				  short_opts, long_opts,
+				  &opt_idx))) {
+		switch (opt) {
+		case 'x':
+			reqc->errcode = EINVAL;
+			snprintf(reqc->line_buf, reqc->line_len,
+					"The 'option' command does not support 'x' or 'xprt'. "
+					"Use the 'listen' command instead.");
+			return reqc->errcode;
+		case 'F':
+		case 'u':
+		case 'V':
+			reqc->errcode = EINVAL;
+			snprintf(reqc->line_buf, reqc->line_len,
+					"The option '%s' must be given at the command line.",
+					argv[optind-1]);
+			return reqc->errcode;
+		default:
+			reqc->errcode = ldmsd_process_cmd_line_arg(opt, optarg);
+			if (reqc->errcode == ENOENT) {
+				snprintf(reqc->line_buf, reqc->line_len,
+					"Unknown cmd-line option or it must be "
+					"given at the command line: %s\n", argv[optind-1]);
+				return reqc->errcode;
+			}
+			break;
+		}
+	}
+	return 0;
+}
+
+static int cmd_line_arg_set_handler(ldmsd_req_ctxt_t reqc)
+{
+	char *s, *token, *ptr1;
+	int rc = 0;
+	int argc = 1;
+	int max_opt = 11;
+	char *argv[11]; /* 11 is the total number of supported cmd-line options in config files */
+
+	s = ldmsd_req_attr_str_value_get_by_id(reqc, LDMSD_ATTR_STRING);
+	opterr = 0;
+
+	argv[0] = "option";
+	for (token = strtok_r(s, " \t\n", &ptr1); token;
+			token = strtok_r(NULL, " \t\n", &ptr1)) {
+		if (argc > max_opt - 2) {
```

Comment:
argv should be big enough to include "option" at the beginning and NULL at the end. Then this can just be "argc >= max_opt". And when that happens, we can just return a "too many options" error.